agent/agent_endpoint_test.go

```diff
@@ -4551,7 +4551,7 @@ func TestAgentConnectCARoots_list(t *testing.T) {
 
 	// Set some CAs. Note that NewTestAgent already bootstraps one CA so this just
 	// adds a second and makes it active.
-	ca2 := connect.TestCAConfigSet(t, a, nil)
+	ca2 := connect.TestCAConfigSet(t, a, nil, connect.DefaultPrivateKeyType, connect.DefaultPrivateKeyBits)
```

Comment:
Did you see my comment on the other PR about undoing all this boilerplate change and reverting `TestCAConfigSet` and `TestCA` to their original signatures? We can add new ones like `TestCAWithKeyType` for the few cases we need the extra flexibility.

That not only makes this PR much smaller but it keeps our test cleaner - we follow the same pattern a lot to make the most common test setup the default and then only add more verbosity when it's really needed.

Reply:
Ooops. My mistake that comment was on https://github.com/hashicorp/consul/pull/6055 which this will inherit when you rebase.

In case you aren't aware (probably are!), if you made this PR off your branch for #6055 you can create the PR with the "base" being that branch instead of master which means we only review changes here not re-review changes in the other PR too!

Reply:
Rebased off #6055 this morning.